app/models/fact.rb

```diff
@@ -15,6 +20,23 @@ class Fact < ApplicationRecord
 
   before_save :update_normalized_type
 
+  scope :birth_dates, -> { where(fact_type: Types::BIRTH_DATE) }
+  scope :birthdays_in_range, lambda { |start_date, end_date, limit|
+    add_valid_date_function
+    date_in_current_year =
+      "make_date(date_part('year', current_date)::int, date_part('month', date_string::date)::int, " \
+      "date_part('day', date_string::date)::int)"
+    includes(:factable)
+      .birth_dates
+      .where(
+        'CASE is_valid_date(date_string) WHEN true THEN ' \
```

Comment:
If I understand it correctly... `is_valid_date` acts as some protection against SQL-injection here right?

Reply:
It will provide some protection against injection, though because it's being casted before it's used I don't think it's overly risky. The main reason this is here is because date string can currently contain strings that SQL is unable to cast to a date, like `1998` or just `random text`. It can handle most formats of date but it has trouble with strings that only specify a month or a year